### ISSUE TITLE
unable to connect to mina daemon : rosetta-v7

### ISSUE THREAD

**Issue by PrernaPadhee** [MinaProtocol/mina#9869]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/MinaProtocol/mina/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/MinaProtocol/mina/discussions

### Description

- I have been trying to setup a mina node using mina-rosetta implementation using https://github.com/MinaProtocol/mina/blob/develop/src/app/rosetta/README.md
- I have opened ports 10101, 3085, 3086, 3087, 8302.
- The current status is that the node is unable to connect with the Mina Daemon and container tends to restart every 30mins-1hr.
![image](https://user-images.githubusercontent.com/29208607/142865814-eed878e0-c15e-4aae-a103-038600e85afd.png)

- It sometimes shows status as bootstrap but cannot attain synced state at any time. 
- On using curl query -> curl --data '{ network_identifier: { blockchain: "mina", network: "mainnet" }, metadata: {} }' 'localhost:1332/network/status'
the response received is "GraphQL query failed"
- i am also observing this log 
![image](https://user-images.githubusercontent.com/29208607/142865921-77d52d81-c1f0-41a6-a636-3661d508182e.png)

### Steps to Reproduce

1. setup mina container using docker image 
2.
3.
...

### Expected Result

mina node should sync with latest block

### Actual Result

unable to connect with mina daemon

### How frequently do you see this issue?

Always

### What is the impact of this issue on your ability to run a node?

Blocker

### Status

```shell
Error: Unable to connect to Mina daemon.
- The daemon might not be running. See logs (in `~/.mina-config/mina.log`) for details under the host:127.0.0.1.
  Run `mina daemon -help` to see how to start daemon.
- If you just started the daemon, wait a minute for the RPC server to start.
- Alternatively, the daemon may not be running the RPC server on (127.0.0.1 8301).
  If so, add flag `---daemon-port` with correct port when running this command.
```

### Additional information

please let me know if there is a tentative date or timeline for the official mina-rosetta docker image release

**Comment by tkdlqm2** [comment]
i have same error 
please help

**Comment by Sventimir** [comment]
I'm investigating this issue. While the daemon seems to connect to the mainnet successfully, it indeed is not reachable by mina client from outside the container.

**Comment by Sventimir** [comment]
I've got trouble reproducing the issue exactly, because `release/1.3.0` is pretty old code and I have trouble compiling the CLI to use from outside the docker, but one thing that I notice is that the client tries to connect to the daemon on port 8301, which is not exposed in the `docker run` command. Please expose port `8301` and try again.

**Comment by Sventimir** [comment]
I'm putting this aside for the time being, as I Have trouble gathering information and there's more urgent work to do on Rosetta.

**Comment by kantp** [comment]
Thanks @Sventimir, that sounds good!

I will move it from "blocked" to the backlog, as it's only blocked on having someone with nothing more urgent and important to do, not on other work that is a prerequiusite.

**Comment by Sventimir** [comment]
@PrernaPadhee, do you still experience this issue? I have trouble building the image, because the issue is quite old. Does the issue appear on other netwroks (devnet, berkeley) too? After the hard fork this configuration will become irrelevant anyway, replaced by newer one, so we wonder if it really makes sense to spend too much time on this issue at this point…

**Comment by robinbb** [comment]
I propose that, with this issue, we should attempt to repro using an updated version of the Mina node software distribution. If it can't be repro'd, then we can mark done, provided there is no more input from the OP.

**Comment by kantp** [comment]
Sounds good. @Sventimir is working on #12452 right now (which he and @ejMina226 discovered while working on #8084), and will return to this later.

**Comment by ejMina226** [comment]
The commands in https://github.com/MinaProtocol/mina/blob/develop/src/app/rosetta/README.md no longer build an image. An error is thrown. @Sventimir is also seeing this.

**Comment by robinbb** [comment]
So, does that means that the instructions in that README.md will be updated (under the auspices of working on this issue), and then the OP asked to repro? (Probably, the OP is gone, frustrated with the experience, and may not expect a reply.)

**Comment by ejMina226** [comment]
@robinbb It looks like the issue may be to do with the processor on the M1 Mac. The build looks to work for other hardware.

**Comment by ejMina226** [comment]
I'm communicating with @kaozenn about this.

**Comment by robinbb** [comment]
Okay, @ejMina226. It would be nice to see, here, what your plan is to close this issue (what you think should be done), after you have that discussion. Thanks for the update.
